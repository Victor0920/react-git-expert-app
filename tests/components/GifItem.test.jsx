import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

const title = 'One Punch Gif';
const url =
    'https://media1.giphy.com/media/l1IYax3ObvxNgOmre/giphy.gif?cid=ecf05e472e77aaeaec8618c3e82a8155b83b1f1fb7d22088&rid=giphy.gif&ct=g';

describe('Tests in <GifItem />', () => {
    test('Must do match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Must show image with url and alt', () => {
        render(<GifItem title={title} url={url} />);

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Must show title in component', () => {
        render(<GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    });
});
