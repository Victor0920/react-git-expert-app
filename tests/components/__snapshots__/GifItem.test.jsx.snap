// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> Must do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Gif"
      src="https://media1.giphy.com/media/l1IYax3ObvxNgOmre/giphy.gif?cid=ecf05e472e77aaeaec8618c3e82a8155b83b1f1fb7d22088&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Gif
    </p>
  </div>
</div>
`;
